### PR TITLE
SaaS: add the ability to set master url

### PR DESCRIPTION
**Pull request by trojikman**

### CODE REVIEW

File: saas/models/saas_operator.py [hunk 1]
```diff
@@ -19,8 +19,9 @@ class SAASOperator(models.Model):
     type = fields.Selection([
         ('local', 'Same Instance'),
     ], 'Type')
-    host = fields.Char()
-    port = fields.Char()
+    master_url = fields.Char(required=True, help='Master URL to manage builds')
```
Comment (yelizariev) [inline saas/models/saas_operator.py]:
```suggestion
    master_url = fields.Char(required=True, help='URL for server-to-server communication ')
```

File: saas_test/tests/test_saas.py [hunk 2]
```diff
@@ -77,11 +77,13 @@ def setUp(self):
         self.saas_operator_1 = self.env['saas.operator'].create({
             'type': 'local',
             'db_url_template': 'http://{db_name}.{db_id}.127.0.0.1.nip.io:8069',
+            'master_url': 'http://saas.127.0.0.1.nip.io:8069',
         })
 
         self.saas_operator_2 = self.env['saas.operator'].create({
             'type': 'local',
             'db_url_template': 'http://{db_name}.{db_id}.127.0.0.1.nip.io:8069',
+            'master_url': 'http://saas.127.0.0.1.nip.io:8069',
```
Comment (yelizariev) [inline saas_test/tests/test_saas.py:86]: "connect to build" doesn't not work on travis with such values

File: saas/data/local_operator.xml [hunk 3]
```diff
@@ -5,5 +5,6 @@
         <field name="name">Same Instance</field>
         <field name="type">local</field>
         <field name="db_url_template">http://{db_name}.{db_id}.127.0.0.1.nip.io</field>
+        <field name="master_url">http://saas.127.0.0.1.nip.io</field>
```
Comment (yelizariev) [inline saas/data/local_operator.xml]: make this computable to value ``http://DBNAME.127.0.0.1.nip.io:8069``

**Review by yelizariev — CHANGES_REQUESTED** [review]
Also, update index.rst about db-filter requirement

File: saas_test/tests/test_saas.py (hunk 2 above)
Comment (yelizariev) [reply saas_test/tests/test_saas.py:86]: Steps for tests:

* get authentication on master database in ``requests``' session
* open "connect to build ulr"
* check that you are finally redirected to ``/web`` page at the build

Comment (yelizariev) [reply saas_test/tests/test_saas.py:86]: For first step use ``authenticate`` method (try to don't copy-paste it) from ``odoo/tests/common.py``

Comment (trojikman) [reply saas_test/tests/test_saas.py:86]: Added TODO to README

File: saas/doc/index.rst [hunk 4]
```diff
@@ -9,6 +9,11 @@ Installation
 * `Install <https://odoo-development.readthedocs.io/en/latest/odoo/usage/install-module.html>`__ this module in a usual way
 * Restart the odoo as required by the Job Queue module
 
+Configuration
+=============
+
+* Use db-filter=^%d$ when working on the same database
```
Comment (yelizariev) [inline saas/doc/index.rst]: Use ``formatting`` for technical stuff.
What the hell is *same database*

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: saas/data/local_operator.xml [hunk 5]
```diff
@@ -5,5 +5,7 @@
         <field name="name">Same Instance</field>
         <field name="type">local</field>
         <field name="db_url_template">http://{db_name}.{db_id}.127.0.0.1.nip.io</field>
+        <!-- It works when db-filter=^%d$ -->
+        <field name="master_url" eval="'http://' + obj().env.cr.dbname + '.127.0.0.1.nip.io:8069'" model="saas.operator"/>
```
Comment (yelizariev) [inline saas/data/local_operator.xml:9]: Rename file to saas_operator_data.xml. See https://www.odoo.com/documentation/12.0/reference/guidelines.html

**Pull request merged**